src/Appwrite/Platform/Tasks/ScheduleBase.php

```diff
@@ -110,105 +154,59 @@ public function action(Group $pools, Database $dbForPlatform, callable $getProje
-                        $regions[] = 'default';
+            foreach ($results as $document) {
+                $localDocument = $this->schedules[$document->getInternalId()] ?? null;
+
+                if ($localDocument !== null) {
+                    if (!$document['active']) {
+                        Console::info("Removing: {$document['resourceType']}::{$document['resourceId']}");
+                        unset($this->schedules[$document->getInternalId()]);
+                    } elseif (strtotime($localDocument['resourceUpdatedAt']) !== strtotime($document['resourceUpdatedAt'])) {
+                        Console::info("Updating: {$document['resourceType']}::{$document['resourceId']}");
+                        $this->schedules[$document->getInternalId()] = $getSchedule($document);
                     }
-
-                    $results = $dbForPlatform->find('schedules', \array_merge($paginationQueries, [
-                        Query::equal('region', $regions),
-                        Query::equal('resourceType', [static::getSupportedResource()]),
-                        Query::greaterThanEqual('resourceUpdatedAt', $lastSyncUpdate),
-                    ]));
-
-                    $sum = count($results);
-                    $total = $total + $sum;
-
-                    foreach ($results as $document) {
-                        $localDocument = $this->schedules[$document->getInternalId()] ?? null;
-
-                        // Check if resource has been updated since last sync
-                        $org = $localDocument !== null ? \strtotime($localDocument['resourceUpdatedAt']) : null;
-                        $new = \strtotime($document['resourceUpdatedAt']);
-
-                        if (!$document['active']) {
-                            Console::info("Removing: {$document['resourceType']}::{$document['resourceId']}");
-                            unset($this->schedules[$document->getInternalId()]);
-                        } elseif ($new !== $org) {
-                            Console::info("Updating: {$document['resourceType']}::{$document['resourceId']}");
-                            $this->schedules[$document->getInternalId()] = $getSchedule($document);
-                        }
+                } else {
+                    try {
+                        $this->schedules[$document->getInternalId()] = $getSchedule($document);
+                    } catch (\Throwable $th) {
+                        $collectionId = static::getCollectionId();
+                        Console::error("Failed to load schedule for project {$document['projectId']} {$collectionId} {$document['resourceId']}");
+                        Console::error($th->getMessage());
                     }
-
-                    $latestDocument = \end($results);
                 }
+            }
 
-                $lastSyncUpdate = $time;
-                $timerEnd = \microtime(true);
-
-                $pools->reclaim();
```

Comment:
the getProjectDB() call consumes one connection from the pool and does not return it back. So it will lead to the pool is empty errors.